### PR TITLE
Match ROCm platform detection in CMake and setup.py. Issue #9690

### PR DESCRIPTION
**Pull request by ipanfilo**
# Description

Update TE package setup.py ROCm platform detection to match logic from TE core CMake

Fixes # (issue)

#9690

- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: build_tools/utils.py [hunk 1]
```diff
@@ -163,6 +163,26 @@ def found_pybind11() -> bool:
...
+    - It is configured with NVTE_USE_ROCM env
+      OR:
+    - HIP compiler is found and CUDA on is not
+    """
+    if os.getenv("NVTE_USE_ROCM"):
+        return True
+
+    try:
+        cuda_path()
+        return True
```
Comment (wangye805) [inline build_tools/utils.py]: If cuda_path() is found, should we return False?

File: build_tools/utils.py [hunk 2]
```diff
@@ -238,6 +258,9 @@ def get_frameworks() -> List[str]:
             _frameworks.extend(arg.replace("--framework=", "").split(","))
             sys.argv.remove(arg)
 
+    if rocm_build():
+        _requested_frameworks = [framework.lower() for framework in _frameworks]
```
Comment (wangye805) [inline build_tools/utils.py:262]: Duplicated with line 292

File: build_tools/utils.py [hunk 3]
```diff
@@ -271,6 +294,31 @@ def get_frameworks() -> List[str]:
         if framework not in supported_frameworks:
             raise ValueError(f"Transformer Engine does not support framework={framework}")
 
+    if rocm_build():
+        _unsupported_frameworks = []
+        if "pytorch" in _frameworks:
+            try:
+                from torch.utils.cpp_extension import IS_HIP_EXTENSION
```
Comment (wangye805) [inline build_tools/utils.py:301]: During docker image building, will this "from torch.utils.cpp_extension import" works

**Review by wangye805 — CHANGES_REQUESTED** [review]
Thanks for working on this Ilya! 

I left some comments. 

In addition, there are some rocm/cuda determination needed in transformer_engine/common dir, for example, https://github.com/ROCm/TransformerEngine/blob/3ce0f87acd7442b1a9059e236139d016b189643f/transformer_engine/common/__init__.py#L105

Can you help address them as well? Or we can leave it for the next PR

File: build_tools/utils.py (hunk 1 above)
Comment (ipanfilo) [reply build_tools/utils.py]: Yes, I mixed up cuda)is_found with use rocm

File: build_tools/utils.py (hunk 2 above)
Comment (ipanfilo) [reply build_tools/utils.py:262]: It is similar but assigns different variable. Indeed, line 292 can be moved here but I try to minimize changes from upstream

File: build_tools/utils.py (hunk 3 above)
Comment (ipanfilo) [reply build_tools/utils.py:301]: It was used in original ROCm detection path, so yes

**Comment by ipanfilo** [comment]
> https://github.com/ROCm/TransformerEngine/blob/3ce0f87acd7442b1a9059e236139d016b189643f/transformer_engine/common/__init__.py#L105
> 
> Can you help address them as well? Or we can leave it for the next PR

Since this is runtime, not build time. We cannot rely on hipcc for ROCm detection. Should we use other logic like env variable but if it is not set try/except to load  CUDA or ROCm libraries ?

**Review by wenchenvincent — APPROVED** [review]

**Pull request merged**